docs/GUIDES.md

```diff
@@ -177,8 +177,242 @@ Our playing field is a grid of tiles, addressable like a coordinate system:
+    state = %{
+      viewport: viewport,
+      width: num_tiles_width,
+      height: num_tiles_height
+    }
+
+    {:ok, state}
+  end
+  ```
+- optionally give it a more Nokia style feel:
+  - default background for the scene is black, but we can set the background like so:
+
+    ```elixir
+    @graph Graph.build(clear_color: :dark_sea_green)
+    ```
+
+- we need to add our graph to the scene to actually draw something
+- add information about game objects: how can we define a snake?
+  - we describe a snake via:
+  - body: a list of ordered x and y pairs. Each pair corresponds to a cell in the grid the snake is currently occupying
+  - size: the current size of the snake in cells
+
+  ```elixir
+  snake = %{body: [{9, 9}], size: 1}
+  ```
+
+- need to update the graph to actually draw something.
+
+  ```elixir
+  # the entire game state will be held here
+  state = %{
+    viewport: viewport,
+    width: num_tiles_width,
+    height: num_tiles_height
+  }
+
+  snake = %{body: [{9, 9}], size: 5}
+
+  # update the graph and push it to be rendered
+  graph =
+    @graph
+    |> draw_object(snake)
+
+  {:ok, state, push: graph}
+  ```
+- need to implement the function to draw our game object onto the scene:
+  - means: we add information to the graph on where to place what
+  - new function: `draw_game_object/2` takes graph and our snake objects (we will have more objects later)
+  - TODO: explain pipelines
+
+  ```elixir
+  defp draw_object(graph, %{body: snake}) do
+    Enum.reduce(snake, graph, fn {x, y}, graph ->
+      draw_tile(graph, x, y, fill: :blue)
+    end)
+  end
+  ```
+
+- match on the object we pass in (the snake)
+- TODO: explain matching on maps
+- snake essentially only a collection of tiles
+- add private function to fill a tile in our coordinate system:
+  - function takes graph and adds the respective rectangles that define our snake
+
+  ```elixir
+  # at top of file
+  import Scenic.Primitives, only: [rrect: 3]
+
+  # group with constants
+  @tile_radius 8
+
+  # draw tiles as rounded rectangles to look nice
+  defp draw_tile(graph, x, y, opts) do
+    tile_opts = Keyword.merge([fill: :white, translate: {x * @tile_size, y * @tile_size}], opts)
+    graph |> rrect({@tile_size, @tile_size, @tile_radius}, tile_opts)
+  end
+  ```
+
+- output: we only see a single dot...
+  - we aren't yet using the size information in our snake
+  - we are just drawing a single dot
+  - reason: our scene is updated only once when the game starts
+    - in order to move, we need update our scene more often (implement the actual game logic)
+- first: look at output: warning that we're not using the args argument:
+
+      warning: variable "arg" is unused (if the variable is not meant to be used, prefix it with an underscore)
+        lib/scenes/game.ex:14: Snake.Scene.Game.init/2
+
+- change this to be ignored intentionally by prepending underscore:
+
+  ```elixir
+  def init(_arg, opts) do
+  ```
```

Comment:
Don't we see the warning already in when running the game in line 206? Maybe better to move this part up?

Reply:
Same here: already done while writing the prose version.